tests/test_registration_widget.py

```diff
@@ -599,6 +603,181 @@ def test_open_auto_slice_dialog_valid(
 
 
+def test__on_auto_slice_parameters_confirmed_starts_worker(
+    registration_widget, mocker
+):
+    params = {
+        "init_points": 3,
+        "n_iter": 5,
+    }
+    total = 2 * (params["init_points"] + params["n_iter"])
+
+    # Mock progress bar
+    progress_bar_mock = mocker.Mock()
+    registration_widget.adjust_moving_image_widget.progress_bar = (
+        progress_bar_mock
+    )
+
+    mocked_worker = mocker.Mock()
+    mocked_create_worker = mocker.patch(
+        "brainglobe_registration.registration_widget.create_worker",
+        return_value=mocked_worker,
+    )
+
+    # Call method
+    registration_widget._on_auto_slice_parameters_confirmed(params)
+
+    # Assert progress bar is set correctly
+    progress_bar_mock.setVisible.assert_called_once_with(True)
+    progress_bar_mock.setValue.assert_called_once_with(0)
+    progress_bar_mock.setRange.assert_called_once_with(0, total)
+
+    # Assert worker created and connected correctly
+    mocked_create_worker.assert_called_once()
+    mocked_worker.yielded.connect.assert_called_once_with(
+        registration_widget.handle_auto_slice_progress
+    )
+    mocked_worker.returned.connect.assert_called_once_with(
+        registration_widget.set_optimal_rotation_params
+    )
+    mocked_worker.start.assert_called_once()
+
+
+def test_strip_ansi_color_filter_removes_escape_codes():
+    record = logging.LogRecord(
+        "name",
+        logging.INFO,
+        "pathname",
+        0,
+        "\x1b[31mRed Text\x1b[0m",
+        (),
+        None,
+    )
+    filt = StripANSIColorFilter()
+    filt.filter(record)
+
+    assert record.msg == "Red Text"
+
+
+def test_run_auto_slice_thread_logs_and_yields_results(
+    mocker, registration_widget
+):
+    atlas_image = np.ones((10, 10), dtype=np.uint8)
+    moving_image = (np.ones((10, 10)) * 2).astype(np.int16)
+
+    mocker.patch(
+        "brainglobe_registration.registration_widget."
+        "get_data_from_napari_layer",
+        side_effect=[atlas_image, moving_image],
+    )
+
+    logging_dir = Path("/tmp/fake_logging_dir")
+    mocker.patch(
+        "brainglobe_registration.registration_widget.get_brainglobe_dir",
+        return_value=logging_dir,
+    )
+
+    mock_args_namedtuple = mocker.Mock()
+    mock_get_args = mocker.patch(
+        "brainglobe_registration.registration_widget."
+        "get_auto_slice_logging_args",
+        return_value=mock_args_namedtuple,
+    )
+
+    mock_start_logging = mocker.patch(
+        "brainglobe_registration.registration_widget.fancylog.start_logging"
+    )
+
+    # Patch logger handlers
+    handler_1 = mocker.Mock()
+    handler_2 = mocker.Mock()
+    handler_1.level = logging.NOTSET
+    handler_2.level = logging.NOTSET
+    logger = logging.getLogger()
+    original_handlers = logger.handlers
+    logger.handlers = [handler_1, handler_2]
+
+    # Spy on logging.info
+    info_spy = mocker.spy(logging, "info")
+
+    final_result = {
```

Comment:
The `final_result` name is overwritten below but it's confusing and if these parameters ever change it might be hard to track down the error. This dict could be `expected_result` and referred as such throughout, then `final_result` is fetched from the end of the iteration and compared to `expected_result`. This way if you ever change the values in `expeced_result` for some reason there's one source of truth.

```suggestion
    expected_result = {
```